src/main/scala/au/com/agiledigital/toolform/tasks/generate/docker/GenerateDockerComposeV3.scala

```diff
@@ -0,0 +1,175 @@
+    if (outFile.isDirectory) {
+      return Left(ToolFormError("Output path is a directory. Docker Compose V3 output requires a single file as an output."))
+    }
+
+    // Logic
+
+    val writer = new BufferedWriter(new FileWriter(outFile, false))
+    try {
+
+      val writeFile = for {
+        _ <- write(s"# Generated by ${BuildInfo.name} (${BuildInfo.version})")
+        _ <- write("version: '3'")
+        _ <- write("services:")
+        i <- indent()
+        _ <- for {
+          _ <- writeEdges(project.id, subEdgeDefsFromProject(project).toList)
+          _ <- writeComponents(project.id, project.sortedComponents.values.toList)
+          _ <- writeResources(project.sortedResources.values.toList)
+        } yield ()
+        _ <- resetIndent(i)
+      } yield ()
+
+      val context = WriterContext(writer)
+      writeFile.eval(context)
+
+      Right("Completed Successfully")
+    } finally {
+      writer.close()
+    }
+  }
+
+  // Sections
+
+  private def writeEdges(projectId: String, subEdgeDefs: List[SubEdgeDef]) = State[WriterContext, Unit] { context =>
+    (subEdgeDefs
+       .traverseU(subEdge => writeSubEdge(projectId, subEdge))
+       .exec(context),
+     ())
+  }
+
+  private def writeComponents(projectId: String, components: List[Component]) = State[WriterContext, Unit] { context =>
+    (components
+       .traverseU(component => writeComponent(projectId, component))
+       .exec(context),
+     ())
+  }
+
+  private def writeResources(resources: List[Resource]) = State[WriterContext, Unit] { context =>
+    (resources
+       .traverseU(resource => writeResource(resource))
+       .exec(context),
+     ())
+  }
+
+  private def writeSubEdge(projectId: String, subEdgeDef: SubEdgeDef) = {
+    val serviceName = subEdgeServiceName(projectId, subEdgeDef)
+    val imageName   = subEdgeImageName(projectId, subEdgeDef)
+    for {
+      _ <- write(s"$serviceName:")
+      i <- indent()
+      _ <- for {
+        _ <- write(s"image: $imageName")
+        _ <- write(s"restart: always")
+        _ <- write(s"ports:")
+        _ <- write(subEdgePortDefinition(subEdgeDef))
+      } yield ()
+      _ <- resetIndent(i)
+    } yield ()
+  }
+
+  private def writeComponent(projectId: String, component: Component) = {
+    val serviceName = componentServiceName(component)
+    val imageName   = componentImageName(projectId, component)
+    for {
+      _ <- write(s"$serviceName:")
+      i <- indent()
+      _ <- for {
+        _ <- write(s"image: $imageName")
+        _ <- write(s"restart: always")
+        _ <- writeComponentLabels(component)
+        _ <- writeEnvironmentVariables(component)
+        _ <- writePorts(component)
+      } yield ()
+      _ <- resetIndent(i)
+    } yield ()
+  }
+
+  private def writeComponentLabels(component: Component) =
+    for {
+      _ <- write("labels:")
+      i <- indent()
+      _ <- for {
+
+        _ <- write(s"source.path: \042${component.path}\042")
```

Comment:
`\"`

Reply:
There is a bug in Scala where it doesn't handle `\"` with string interpolation!

This was the workaround I used: https://stackoverflow.com/a/39457924